comid/rawint.go

```diff
@@ -0,0 +1,330 @@
+
+// Valid is a no-op for RawIntInteger and always returns nil
+func (o RawIntInteger) Valid() error { return nil }
+
+// String converts RawIntInteger into a string
+func (o RawIntInteger) String() string { return fmt.Sprint(int64(o)) }
+
+// Type returns the name/type of RawIntInteger
+func (o RawIntInteger) Type() string {
+	return RawIntIntegerType
+}
+
+// CompareAgainstRefInteger compares RawIntInteger against another RawIntInteger reference value
+func (o RawIntInteger) CompareAgainstRefInteger(ref RawIntInteger) bool {
+	return o == ref
+}
+
+// CompareAgainstRefRange compares RawIntInteger against a TaggedRawIntRange
+func (o RawIntInteger) CompareAgainstRefRange(ref TaggedRawIntRange) bool {
+	obj, err := convertRawIntIntegerToInt64(o)
+	if err != nil {
+		log.Printf("RawIntInteger:CompareAgainstRefRange: Error: %v", err)
+		return false
+	}
+
+	if ref.Min != nil && obj < *ref.Min {
+		return false
+	}
+
+	if ref.Max != nil && obj > *ref.Max {
+		return false
+	}
+
+	return true
+}
+
+func convertRawIntIntegerToInt64(val any) (int64, error) {
+	switch v := val.(type) {
+	case RawIntInteger:
+		return int64(v), nil
+	case *RawIntInteger:
+		return int64(*v), nil
+	default:
+		return 0, fmt.Errorf("unexpected type for RawIntInteger: %T", v)
+	}
+}
+
+// TaggedRawIntRangeType is the name of the Range version of RawInt
+const TaggedRawIntRangeType = "rawIntRange"
+
+// TaggedRawIntRange implements the Range version of RawInt. The range is
+// made of minimum and maximum values. If the minimum is nil, it's assumed
+// to be negative infinity. If the maximum is nil, it's assumed to be
+// positive infinity.
+type TaggedRawIntRange struct {
+	Min *int64
+	Max *int64
+}
+
+// NewRawIntRange creates a TaggedRawIntRange with the input value
+func NewRawIntRange(val any) (*TaggedRawIntRange, error) {
+	var ret TaggedRawIntRange
+
+	if val == nil {
+		return &ret, nil
+	}
+
+	switch v := val.(type) {
+	case TaggedRawIntRange:
+		ret = v
+	case *TaggedRawIntRange:
+		ret = *v
+	default:
+		return nil, fmt.Errorf("unexpected type for TaggedRawIntRange: %T", v)
+	}
+
+	return &ret, nil
+}
+
+// Valid checks if TaggedRawIntRange is valid
+func (o TaggedRawIntRange) Valid() error {
+	if o.Min != nil && o.Max != nil && *o.Min > *o.Max {
+		return fmt.Errorf("TaggedRawIntRange: Invalid Range, Min: %d Max: %d", *o.Min, *o.Max)
+	}
+
+	return nil
+}
+
+// String converts TaggedRawIntRange to a string
+func (o TaggedRawIntRange) String() string {
+	var rangeMin, rangeMax string
+
+	if o.Min != nil {
+		rangeMin = fmt.Sprintf("[%d", *o.Min)
+	} else {
+		rangeMin = "(-inf"
+	}
+
+	if o.Max != nil {
+		rangeMax = fmt.Sprintf("%d]", *o.Max)
+	} else {
+		rangeMax = "inf)"
+	}
+
+	return fmt.Sprintf("%s:%s", rangeMin, rangeMax)
+}
+
+// Type returns the name/type of TaggedRawIntRange
+func (o TaggedRawIntRange) Type() string {
+	return TaggedRawIntRangeType
+}
+
+// CompareAgainstRefInteger compares TaggedRawIntRange against a given
+// RawIntInteger reference
```

Comment:
This should describe the comparison rule. An int range is equal to an int iff the min and max are both equal to the int.